### PR TITLE
Feat/user profile page

### PR DESCRIPTION
**Pull request by dimonl**
new PR remastered

### CODE REVIEW

File: frontend/src/app/libs/user-profile-page/model/user-dto.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface UserDto {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/model/user-dto.ts:1]: seems like this interface already exists in application, please use existing one

File: frontend/src/app/libs/user-profile-page/model/user.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface IUser {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/model/user.ts:1]: also this interface already exists

File: frontend/src/app/libs/user-profile-page/service/user-repository.service.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+const URL_PREFIX: string = '/api';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class UserRepository {
+	constructor(private http: HttpClient) {}
+
+	public updateUser(user: Partial<User>): Observable<any> {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/service/user-repository.service.ts:14]: avoid any

File: frontend/src/app/libs/user-profile-page/service/user-service.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+				return throwError(new Error('Сервер не отвечает. Проверьте подключение к сети.'));
+			}
+
+			console.error(`Backend returned code ${error.status}, ` + `body was: ${error.error}`);
+		}
+
+		return throwError(new Error('Ошибка сервера. Повторите запрос позже.'));
+	}
+
+	public updateUser(user: Partial<IUser>): Observable<any> {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/service/user-service.ts:39]: this method must be placed in repository

File: frontend/src/app/libs/user-profile-page/service/user-service.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+const URL_PREFIX: string = '/api';
+const ERROR_STATUS: number = 504;
+
+@Injectable({
+	providedIn: 'root',
+})
+export class UserService {
+	constructor(private http: HttpClient) {}
+
+	private handleError(error: HttpErrorResponse): any {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/service/user-service.ts:16]: this can be removed

File: frontend/src/app/libs/user-profile-page/store/user/user.actions.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import { Action } from '@ngrx/store';
+import { User } from '../../../auth/models/user';
+import { UserChanges } from '../../model/user-changes';
+
+export enum UserActionTypes {
+	GetUserFromAuth = '[User] Get user from auth',
+	GetUserFromAuthSuccess = '[User] Get user from auth success',
+	GetUserFromAuthFail = '[User] Get user from auth failed',
+	UpdateUser = '[User] Update user',
+	UpdateUserInfoSuccessfuly = '[User] Update user success',
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.actions.ts:10]: ...Success, please check Event naming convention at wiki

File: frontend/src/app/libs/user-profile-page/store/user/user.actions.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+	UpdateUserInfoFailed = '[User] Update user failed',
+}
+
+export class GetUserFromAuthAction implements Action {
+	public readonly type: UserActionTypes = UserActionTypes.GetUserFromAuth;
+}
+
+export class GetUserFromAuthSuccessfullyAction implements Action {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.actions.ts:18]: ALso here ...Success

File: frontend/src/app/libs/user-profile-page/store/user/user.effects.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import { User } from '../../../auth/models/user';
+import { UserConverter } from '../../service/user-converter.service';
+import { UserDto } from '../../model/user-dto';
+
+@Injectable()
+export class UserEffects {
+	public getUser$: Observable<UserActionType> = createEffect(() =>
+		this.actions$.pipe(
+			ofType(UserActionTypes.GetUserFromAuth),
+			// map((action: GetUserFromAuthAction) => action),
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.effects.ts:26]: remove commented code

File: frontend/src/app/libs/user-profile-page/store/user/user-state.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+import { EntityWrapper } from '../../../auth/models/entity-wraper';
+import { User } from '../../../auth/models/user';
+
+export interface UserState {
+	userData: EntityWrapper<User>;
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user-state.ts:5]: just `user`

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+			email: [''],
+			password: [''],
+			mobilePhone: [''],
+			firstName: [''],
+			secondName: [''],
+		});
+	}
+
+	private navigateFromUserProfile(): void {
+		this.router.navigate(['/']);
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:32]: no need to navigate somewhere

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+			firstName: this.profileForm.value.firstName,
+			secondName: this.profileForm.value.secondName,
+			mobilePhone: this.profileForm.value.mobilePhone,
+		};
+		return valuesFromForm;
+	}
+
+	public ngOnInit(): void {
+		this.userFacadeService.getUserFromAuth();
+		this.userFacadeService.user$.pipe(takeUntil(this.destroySource)).subscribe((user: EntityWrapper<User>) => {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:60]: takeUntil(this.destroy$) - follow name convention

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+			email: this.profileForm.value.email,
+			password: this.profileForm.value.password,
+			firstName: this.profileForm.value.firstName,
+			secondName: this.profileForm.value.secondName,
+			mobilePhone: this.profileForm.value.mobilePhone,
+		};
+		return valuesFromForm;
+	}
+
+	public ngOnInit(): void {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:58]: public methods must be above private

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+		});
+		if (this.userForm !== null) {
+			this.setValuesToForm();
+		}
+	}
+
+	public updateUserInfo(): void {
+		const formValue: User = this.getChangesFromForm();
+		this.userFacadeService.updateUser(formValue);
+		this.userForm = null;
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:73]: not use we need to remove whole form object when just updating user value

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+		this.navigateFromUserProfile();
+	}
+
+	public ngOnDestroy(): void {
+		this.destroySource.next(true);
+		this.destroySource.complete();
+	}
+
+	public backFromUserProfilePage(): void {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:82]: no need to go somewhere, cancel button can be implemented my just storing original user value somewhere

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: frontend/src/app/libs/user-profile-page/service/user-service.ts (hunk 5 above)
Comment (clain-atomi) [reply frontend/src/app/libs/user-profile-page/service/user-service.ts:16]: avoid any

File: frontend/src/app/libs/user-profile-page/service/user-service.ts (hunk 4 above)
Comment (clain-atomi) [reply frontend/src/app/libs/user-profile-page/service/user-service.ts:39]: avoid any

File: frontend/src/app/libs/user-profile-page/store/user/user.actions.spec.ts [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+				type: UserActionTypes.UpdateUserInfoSuccessfuly,
+				payload: { user },
+			});
+		});
+	});
+
+	describe('update User Failure', () => {
+		it('should create UpdateUserInfoFailedAction action', () => {
+			const message: string = 'testError';
+			const err: any = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.actions.spec.ts:98]: avoid any

File: frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+			],
+		});
+		mockUserService = TestBed.inject(UserRepository) as jasmine.SpyObj<UserRepository>;
+		mockAuthFacadeService = TestBed.inject(AuthFacadeService) as jasmine.SpyObj<AuthFacadeService>;
+		mockUserConverter = TestBed.inject(UserConverter) as jasmine.SpyObj<UserConverter>;
+	});
+
+	describe('update user success', () => {
+		beforeEach(() => {
+			const payload: any = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts:66]: avoid any

File: frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+		});
+
+		it('should success', () => {
+			expect(createEffects(actions).updateUser$).toBeObservable(expected);
+		});
+	});
+
+	describe('update user error', () => {
+		beforeEach(() => {
+			const payload: any = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts:81]: avoid any

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+	UserActionType,
+} from './user.actions';
+import { UserState } from './user-state';
+import { EntityStatus } from '../../../auth/models/entity-status';
+import { User } from '../../../auth/models/user';
+
+describe('UserReducers', () => {
+	describe('Init', () => {
+		it('should return the initial state', () => {
+			const action: any = {} as any;
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts:18]: avoid any

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+			expect(newState.userData).toEqual({
+				status: EntityStatus.Success,
+				value: user,
+			});
+		});
+	});
+
+	describe('GetUserFromAuthFailed', () => {
+		it('should return the user.status - true', () => {
+			const err: any = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts:54]: avoid any

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+			expect(newState.userData).toEqual({
+				status: EntityStatus.Success,
+				value: user,
+			});
+		});
+	});
+
+	describe('UpdateUserInfoFailed', () => {
+		it('should return error', () => {
+			const err: any = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts:109]: avoid any

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts [hunk 21]
```diff
@@ -0,0 +1,140 @@
...
+			password: 'test',
+			firstName: 'test',
+			secondName: 'test',
+			mobilePhone: 'test',
+			id: 'test',
+			isAdmin: false,
+		},
+	};
+
+	const updateUser: any = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:27]: avoid any

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+		mobilePhone: 'test',
+		_id: 'test',
+	};
+
+	let component: UserProfilePageComponent;
+	let destroy$: Subject<boolean>;
+	let userForm: User;
+	let fixture: ComponentFixture<UserProfilePageComponent>;
+	const mockUserFacadeService: any = {
+		get user$(): any {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:39]: avoid any

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+		beforeEach(() => {
+			destroy$ = new Subject<boolean>();
+		});
+
+		describe('get auth user', () => {
+			beforeEach(() => {
+				mockUserFacadeService.user$.pipe(takeUntil(destroy$)).subscribe((users: any) => {
+					userForm = users.value;
+				});
+				spyOn<any>(component, 'setValuesToForm').and.stub();
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:87]: you can use non generic spyOn here

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts [hunk 24]
```diff
@@ -0,0 +1,140 @@
...
+			component.backFromUserProfilePage();
+		});
+		it('should redirect', () => {
+			expect(component['router'].navigate).toHaveBeenCalledWith(['/']);
+		});
+	});
+
+	describe('backFromUserProfilePage', () => {
+		beforeEach(() => {
+			spyOn(component as any, 'navigateFromUserProfile').and.callThrough();
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:122]: avoid any

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: frontend/src/app/libs/user-profile-page/model/user.ts (hunk 2 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/model/user.ts:1]: Killed

File: frontend/src/app/libs/user-profile-page/service/user-service.ts (hunk 4 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/service/user-service.ts:39]: killed

File: frontend/src/app/libs/user-profile-page/service/user-service.ts (hunk 5 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/service/user-service.ts:16]: user-service has been killed

File: frontend/src/app/libs/user-profile-page/store/user/user.actions.ts (hunk 6 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.actions.ts:10]: done

File: frontend/src/app/libs/user-profile-page/store/user/user.actions.ts (hunk 7 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.actions.ts:18]: and here...

File: frontend/src/app/libs/user-profile-page/store/user/user-state.ts (hunk 9 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user-state.ts:5]: done

File: frontend/src/app/libs/user-profile-page/store/user/user.effects.ts (hunk 8 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.effects.ts:26]: removed

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts (hunk 13 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:73]: removed

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts (hunk 11 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:60]: done

File: frontend/src/app/libs/user-profile-page/service/user-repository.service.ts (hunk 3 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/service/user-repository.service.ts:14]: done

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts (hunk 21 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:27]: done

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts (hunk 10 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:32]: removed

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts (hunk 14 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:82]: removed

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts (hunk 18 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts:18]: done

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts (hunk 24 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:122]: removed test

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts (hunk 22 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:39]: done

File: frontend/src/app/libs/user-profile-page/store/user/user.actions.spec.ts (hunk 15 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.actions.spec.ts:98]: done

File: frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts (hunk 17 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts:81]: its test is not needed - deleted

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts (hunk 19 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts:54]: done

File: frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts (hunk 16 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.effects.spec.ts:66]: done

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts (hunk 20 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.reducer.spec.ts:109]: done

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts (hunk 12 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:58]: maybe: private - first, public - after?

File: frontend/src/app/libs/user-profile-page/model/user-dto.ts (hunk 1 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/model/user-dto.ts:1]: nope

Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/model/user-dto.ts:1]: there is no interface to update changed user fields

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts (hunk 23 above)
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.spec.ts:87]: its one of the way to do test of private method, 
https://stackoverflow.com/questions/17885635/private-method-unit-testing-with-jasmine/38144942

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 25]
```diff
@@ -0,0 +1,79 @@
...
+
+@Component({
+	selector: 'app-user-config',
+	templateUrl: './user-profile-page.component.html',
+	styleUrls: ['./user-profile-page.component.scss'],
+})
+export class UserProfilePageComponent implements OnInit, OnDestroy {
+	private destroy$: Subject<boolean> = new Subject<boolean>();
+	public profileForm: FormGroup;
+	public userForm: User;
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:19]: just `user`

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts (hunk 12 above)
Comment (clain-atomi) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:54]: no, public - first

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 26]
```diff
@@ -0,0 +1,79 @@
...
+	}
+
+	public ngOnInit(): void {
+		this.userFacadeService.getUserFromAuth();
+		this.userFacadeService.user$.pipe(takeUntil(this.destroy$)).subscribe((user: EntityWrapper<User>) => {
+			if (user.status === EntityStatus.Success) {
+				this.userForm = user.value;
+			}
+		});
+		if (this.userForm !== null) {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:61]: seems like this `if` must be inside `.user$` subscription

Comment (clain-atomi) [reply frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:61]: you can just set value to form inside subscription

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 27]
```diff
@@ -0,0 +1,79 @@
...
+		const formValue: User = this.getChangesFromForm();
+		this.userFacadeService.updateUser(formValue);
+	}
+
+	public ngOnDestroy(): void {
+		this.destroy$.next(true);
+		this.destroy$.complete();
+	}
+
+	public backFromUserProfilePage(): void {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:76]: why do we need empty method which return null? Seems like you can remove it

File: frontend/src/app/libs/user-profile-page/service/user-converter.service.spec.ts [hunk 28]
```diff
@@ -0,0 +1,68 @@
...
+		email: '1@1.com',
+		isAdmin: false,
+		id: 'test',
+		firstName: 'undefined',
+		mobilePhone: 'undefined',
+		secondName: 'undefined',
+		password: 'undefined',
+	};
+
+	const userChanges: UserChanges = {
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/service/user-converter.service.spec.ts:33]: seems like you can use just `User` interface here, add few necessary fields and all will be fine

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: frontend/src/app/libs/user-profile-page/store/user/user.reducer.ts [hunk 29]
```diff
@@ -0,0 +1,81 @@
...
+					value: (action as UpdateUserAction).payload.user,
+				},
+			};
+		}
+		case UserActionTypes.UpdateUserInfoSuccess: {
+			return {
+				...state,
+				user: {
+					status: EntityStatus.Success,
+					value: (action as UpdateUserInfoSuccessAction).payload.user,
```
[the remark this answers is not in the document]
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/store/user/user.reducer.ts:64]: no, and I add it to auth store reducers.

File: frontend/src/app/libs/user-profile-page/user-profile-page-routing.module.ts [hunk 30]
```diff
@@ -0,0 +1,11 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+import { UserProfilePageComponent } from './user-profile-page.component';
+
+const routes: Routes = [{ path: '', component: UserProfilePageComponent }];
```
[the remark this answers is not in the document]
Comment (dimonl) [reply frontend/src/app/libs/user-profile-page/user-profile-page-routing.module.ts:6]: my route is created in lazy loading description

File: frontend/src/app/libs/auth/services/auth-converter.service.ts [hunk 31]
```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { User } from '../models/user';
+import { UserDto } from '../models/user-dto';
+import { AuthResponse } from '../models/auth-response';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class AuthConverter {
+	public convertFromAuthResponse(authResponse: AuthResponse): User {
```
Comment (clain-atomi) [inline frontend/src/app/libs/auth/services/auth-converter.service.ts:10]: No need to pass whole auth reponse here, pass just userDto

Comment (clain-atomi) [reply frontend/src/app/libs/auth/services/auth-converter.service.ts:10]: Seems like this method is unnecessary

File: frontend/src/app/libs/auth/store/auth/auth.effects.ts [hunk 32]
```diff
@@ -1,79 +1,104 @@
...
+	public SignIn: Observable<Action> = this.actions.pipe(
+		ofType<SignIn>(AuthActionTypes.SignIn),
+		exhaustMap((action: SignIn) => {
+			return from(this.fireAuthService.auth.signInWithEmailAndPassword(action.payload.email, action.payload.password));
+		}),
+		map((userModel: auth.UserCredential) => userModel.user),
+		switchMap((firebaseUser: firebase.User) => {
+			return this.authRepository.signIn(firebaseUser.email, firebaseUser.uid).pipe(
+				map((authResponse: AuthResponse) => {
+					return new SignInSuccess(this.authConverter.convertFromAuthResponse(authResponse));
```
Comment (clain-atomi) [inline frontend/src/app/libs/auth/store/auth/auth.effects.ts:38]: use .fromDto(authResponse.user)

File: frontend/src/app/libs/auth/store/auth/auth.effects.ts [hunk 33]
```diff
@@ -1,79 +1,104 @@
...
+	public SignInByGithub: Observable<Action> = this.actions.pipe(
+		ofType<SignInByGithub>(AuthActionTypes.SignInByGithub),
+		exhaustMap(() => {
+			return from(this.fireAuthService.auth.signInWithPopup(new auth.GithubAuthProvider()));
+		}),
+		map((userModel: auth.UserCredential) => userModel.user),
+		switchMap((firebaseUser: firebase.User) => {
+			return this.authRepository.signInByGithub(firebaseUser.email, firebaseUser.uid).pipe(
+				map((authResponse: AuthResponse) => {
+					return new SignInSuccess(this.authConverter.convertFromAuthResponse(authResponse));
```
Comment (clain-atomi) [inline frontend/src/app/libs/auth/store/auth/auth.effects.ts:57]: same here

File: frontend/src/app/libs/auth/store/auth/auth.reducer.spec.ts [hunk 34]
```diff
@@ -98,4 +110,59 @@ describe('AuthReducers', () => {
...
+				firstName: '1',
+				secondName: '1',
+				mobilePhone: '1',
+				password: '1',
+				isAdmin: false,
+				id: '0',
+				email: 'test@test.test',
+			};
+			const action: AuthActions = new UpdateUserInfoSuccess({ user });
+			const newState: AuthState = authReducer(initialState, action);
```
Comment (clain-atomi) [inline frontend/src/app/libs/auth/store/auth/auth.reducer.spec.ts:146]: pre requisites for `it` must be placed in `beforeEach`

Comment (clain-atomi) [reply frontend/src/app/libs/auth/store/auth/auth.reducer.spec.ts:146]: please check all test and fix this

File: frontend/src/app/libs/auth/store/auth/auth.reducer.ts [hunk 35]
```diff
@@ -65,8 +72,8 @@ export function authReducer(state: AuthState = initialState, action: AuthActions
 				...state,
 				user: {
 					status: EntityStatus.Success,
-					value: action.payload,
-				}
+					value: (action as SignUpSuccess).payload,
+				},
```
Comment (clain-atomi) [inline frontend/src/app/libs/auth/store/auth/auth.reducer.ts:76]: no need to add `,` everywhere, please revert this change

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.html [hunk 36]
```diff
@@ -0,0 +1,45 @@
...
+
+    <div>
+        <div>
+            <button mat-raised-button (click)="updateUserInfo()">Save changes</button>
+        </div>
+        <div>
+            <button mat-raised-button>Cancel</button>
+        </div>
+    </div>
+</form>
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.html:45]: please add new line at the end of file

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 37]
```diff
@@ -0,0 +1,74 @@
...
+			firstName: this.user.firstName,
+			secondName: this.user.secondName,
+			mobilePhone: this.user.mobilePhone,
+		});
+	}
+
+	private getChangesFromForm(): User {
+		const valuesFromForm: User = {
+			id: this.user.id,
+			isAdmin: false,
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:44]: lets avoid this line, i mean here you can just store and modify `User` instance inside this form, and pass updated version of current user to `.updateUser()`

File: frontend/src/app/libs/user-profile-page/user-profile-page.component.ts [hunk 38]
```diff
@@ -0,0 +1,74 @@
...
+				this.user = user.value;
+			}
+			if (this.user !== null) {
+				this.setValuesToForm();
+			}
+		});
+	}
+
+	public updateUserInfo(): void {
+		const formValue: User = this.getChangesFromForm();
```
Comment (clain-atomi) [inline frontend/src/app/libs/user-profile-page/user-profile-page.component.ts:66]: more meaningfull name - `updatedUser`

**Review by clain-atomi — CHANGES_REQUESTED** [review]

File: frontend/src/app/libs/auth/store/auth/auth.reducer.ts [hunk 39]
```diff
@@ -7,79 +15,99 @@ export const authFeatureKey: string = 'auth';
...
 		}
-
 		case AuthActionTypes.SignUpSuccess: {
 			return {
 				...state,
 				user: {
 					status: EntityStatus.Success,
-					value: action.payload,
-				}
+					value: (action as SignUpSuccess).payload,
```
Comment (clain-atomi) [inline frontend/src/app/libs/auth/store/auth/auth.reducer.ts:70]: can we avoid `action as`?
Also please remove unnecessary `,` in this file

**Review by clain-atomi — CHANGES_REQUESTED** [review]
